### PR TITLE
New chain (preliminary): ROI finder + seeding + clustering

### PR DESCRIPTION
**Pull request by bfonta**
This PR brings a first version of a ROI-based reconstruction chain. Much work is still required, but this aims to provide a common ground from which to develop further algorithms. It includes:
- ROI finder by @mchiusi 
- seeding on U/V space
- additional plotting for preliminary seeding studies
- clustering on the (x/z, y/z) space
- scripts for validation and final plotting

The new chain (```bye_splits/scripts/run_default_chain.py```) works in the same fashion as the now "default" chain (```bye_splits/scripts/run_default_chain.py```). 

**Minor changes**

- Rename ```data_handle.py``` to ```data_process.py``` to avoid confusion with the name of its module
- Completely remove ```Path``` to make a consistent usage of ```os.path```
- Move common selection cuts to ```bye_splits/data_handle/data_process.baseline_selection(...)```

### CODE REVIEW

File: bye_splits/tasks/roi.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    uniqueu = roi_tcs.tc_wu.unique()
+    uniquev = roi_tcs.tc_wv.unique()
+    with open(params.CfgPath, 'r') as afile:
+        cfg = yaml.safe_load(afile)
+        ncee = cfg['geometry']['nlayersCEE']
+    return ev_tc[(ev_tc.tc_wu.isin(uniqueu)) & (ev_tc.tc_wv.isin(uniquev)) &
+                 (ev_tc.tc_layer <= ncee)]
+
+    
+def roi(pars, df_gen, df_cl, df_tc, **kw):
```
Comment (isehle) [inline bye_splits/tasks/roi.py:40]: I assume this should have been deleted when the function was written (line 73)?

**Review by isehle — COMMENTED** [review]
Overall this looks great, only saw a couple things that might need a quick change.

File: bye_splits/tasks/seed.py [hunk 2]
```diff
@@ -103,7 +103,7 @@ def seed(pars, debug=False, **kw):
                 print('NSeeds={}\tMipPt={}\tX={}\tY={}'.format(len(res[0]),res[0],res[1],res[2])) 
      
             storeOut[key] = res
-            storeOut[key].attrs['columns'] = ['seedEn', 'seedX', 'seedY']
+            storeOut[key].attrs['coliumns'] = ['seedEn', 'seedXdivZ', 'seedYdivZ']
```
[the remark this answers is not in the document]
Comment (bfonta) [reply bye_splits/tasks/seed.py:106]: Fixed in the next commit.

File: bye_splits/tasks/roi.py (hunk 1 above)
Comment (bfonta) [reply bye_splits/tasks/roi.py:40]: This is just a placeholder for when @mchiusi develops other ROI algorithms. I've added a comment.

**Review by isehle — APPROVED** [review]
Seems good to me!

File: bye_splits/tasks/seed_roi.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+import bye_splits
+from bye_splits.utils import common, params
+
+import re
+import numpy as np
+import h5py
+import pandas as pd
+
+def calc_universal_coordinates(df, varu='univ_u', varv='univ_v'):
```
Comment (mchiusi) [inline bye_splits/tasks/seed_roi.py:18]: Which is the purpose of the this function for your seeding stage? What are the _universal u,v coordinates_?

File: bye_splits/plot/roi_uv_space.py [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+from bokeh.layouts import layout
+from bokeh import models as bmd
+from bokeh.plotting import figure
+from bokeh.palettes import viridis as _palette
+from bokeh.util.hex import axial_to_cartesian
+
+import utils
+from utils import params, common, parsing
+
+def calc_universal_coordinates(df, varu='univ_u', varv='univ_v'):
```
Comment (mchiusi) [inline bye_splits/plot/roi_uv_space.py:23]: This function is the same in `seed_roi.py`. Do you want to call it from here (or viceversa), or keep both for now?

File: bye_splits/plot/roi_uv_space.py [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+if __name__ == '__main__':
+    import argparse
+    parser = argparse.ArgumentParser(description='Study ROI UV/grid distributions')
+    parser.add_argument('--fullroi', action='store_true',
+                        help='display an toy full region of interest')
+    parsing.add_parameters(parser) 
+    FLAGS = parser.parse_args()
+ 
+    seed_d = params.read_task_params('seed_roi')
+    roi_event_loop(vars(FLAGS), **seed_d)
```
Comment (mchiusi) [inline bye_splits/plot/roi_uv_space.py:236]: Can you briefly comment which is the main aim of this `.py` file and why you need it in this `plot` folder?

**Review by mchiusi — APPROVED** [review]
I leave just general comments.
Well done. Curious to see the new _ROI finder_ and the _seeding_ algorithms running together!

File: bye_splits/plot/roi_uv_space.py (hunk 4 above)
Comment (bfonta) [reply bye_splits/plot/roi_uv_space.py:23]: Let us keep them separate for now, as I am not sure what will be needed in the plotting later on. But you are right, if they turn out to be identical there should be only one.

File: bye_splits/plot/roi_uv_space.py (hunk 5 above)
Comment (bfonta) [reply bye_splits/plot/roi_uv_space.py:236]: It produces plots that allows to visualize the u/v space in standard cartesian coordinates. It is useful for understanding the  seeding in the cell u/v space.

**Pull request merged**